src/views/collections/CreateCollection/CreateCollection.vue

```diff
@@ -0,0 +1,177 @@
+          </div>
+          <div class="col-md-6">
+            <CollectionPuzzles
+                          :puzzlelist="puzzlelist"
+              @update:puzzle="(puzzles) => (puzzlelist = puzzles)"
+            />
+          </div>
+        </div>
+        <div class="row">
+          <div class="col-md-6">
+            <b-button type="submit" variant="primary" @click="submit()">
+              {{ $t('create-collection:collection-info:main-action') }}
+            </b-button>
+          </div>
+        </div>
+      </div>
+    </div>
+  </EternaPage>
+</template>
+
+<script lang="ts">
+// @ts-ignore
+  import debounce from 'lodash.debounce';
+  import {
+    Component,
+    Vue,
+    Mixins,
+    Watch,
+    Ref,
+    Prop,
+  } from 'vue-property-decorator';
+  import { RouteCallback, Route } from 'vue-router';
+  import axios, { AxiosInstance } from 'axios';
+  // @ts-ignore
+  import VueBootstrapTypeahead from 'vue-bootstrap-typeahead';
+  import draggable from 'vuedraggable';
+  import EternaPage from '@/components/PageLayout/EternaPage.vue';
+  import TagsPanel from '@/components/Sidebar/TagsPanel.vue';
+  import Utils from '@/utils/utils';
+  import { PuzzleItem } from '@/types/common-types';
+  import CollectionInfo from './components/CollectionInfo.vue';
+  import CollectionPuzzles from './components/CollectionPuzzles.vue';
+  import LabViewData, { LabData } from './types';
+
+  @Component({
+    components: {
+      EternaPage,
+      TagsPanel,
+      CollectionInfo,
+      CollectionPuzzles,
+    },
+  })
+  export default class CreateCollection extends Vue {
+    private title = '';
+
+    private body = '';
+
+    private targetName = '';
+
+    private puzzlenames: PuzzleItem[] = [];
+
+    private puzzlelist: PuzzleItem[] = [];
+
+    private currentPicture?: string;
+
+    private newPicture: File | null = null;
+
+    private idInput: String = '';
+
+    get picture() {
+      if (this.newPicture) {
+        return URL.createObjectURL(this.newPicture);
+      }
+      return Utils.getCollectionAvatar(this.currentPicture || null);
+    }
+
+    @Prop({ required: true }) private loading!: boolean;
+
+    async submit() {
+      this.loading = true;
+      const data = new FormData();
+      data.set('collection-title-input', this.title);
+      data.set('collection-description-input', this.body);
+      const puzzleids: String[] = [];
+      this.puzzlelist.forEach((e) => puzzleids.push(e.id));
+      data.set('collection-puzzles', puzzleids.toString());
+      if (this.newPicture) data.append(`files[picture_upload]`, this.newPicture);
+      data.set('type', 'create_collection');
+
+      try {
+        const res = await this.$http.post('/post/', data, {
+          headers: {
+            'Content-type': 'multipart/form-data',
+          },
+        });
+        this.loading = false;
+        const error = res?.data?.data?.error;
+        if (error) throw new Error(error);
+        this.$router.push(`/collections/`);
```

Comment:
Do we want to go to /collections or do we want to go to the page for this specific collection?